packages/compass-sidebar/src/components/use-filtered-connections.ts

```diff
@@ -341,20 +339,63 @@ type UseFilteredConnectionsHookResult = {
     databaseId: string,
     isExpanded: boolean
   ): void;
-  onConnectionsChanged(this: void, newConnections: ConnectionInfo[]): void;
 };
 
-export const useFilteredConnections = (
-  connections: SidebarConnection[],
-  filterRegex: RegExp | null,
-  _fetchAllCollections: () => void,
-  _onDatabaseExpand: (connectionId: string, databaseId: string) => void
-): UseFilteredConnectionsHookResult => {
+function filteredConnectionsToSidebarConnection(
+  filteredConnections: FilteredConnection[]
+): SidebarConnection[] {
+  const sidebarConnections: SidebarConnection[] = [];
+  for (const connection of filteredConnections) {
+    if (connection.connectionStatus === ConnectionStatus.Connected) {
+      sidebarConnections.push({
+        ..._.omit(connection, ['isMatch']),
+        databases: connection.databases.map((database) => {
+          return {
+            ..._.omit(database, ['isMatch']),
+            collections: database.collections.map((collection) =>
+              _.omit(collection, ['isMatch'])
+            ),
+          };
+        }),
+      });
+    } else {
+      sidebarConnections.push({
+        ..._.omit(connection, ['isMatch']),
+      });
+    }
+  }
+  return sidebarConnections;
+}
+
+export const useFilteredConnections = ({
+  connections,
+  filterRegex,
+  fetchAllCollections,
+  onDatabaseExpand,
+}: {
+  connections: SidebarConnection[];
+  filterRegex: RegExp | null;
+  fetchAllCollections: () => void;
+  onDatabaseExpand: (connectionId: string, databaseId: string) => void;
+}): UseFilteredConnectionsHookResult => {
   const [{ filtered, expanded }, dispatch] = useReducer(connectionsReducer, {
     filtered: undefined,
     expanded: {},
   });
 
+  const activeConnections = useMemo(() => {
+    return connections
+      .filter(({ connectionStatus }) => {
+        return connectionStatus === ConnectionStatus.Connected;
+      })
+      .map(({ connectionInfo }) => connectionInfo);
+  }, [connections]);
+
+  // get rid of stale connection related metadata in the state
```

Comment:
this is the change that fixes the bug - the rest of the changes are based on the tests that I added.